libs/ledger-live-common/src/exchange/swap/mock.ts

```diff
@@ -153,6 +176,56 @@ export const mockGetProviders: GetProviders = async () => {
     },
   ];
 };
+
+// Providers using V4 schema. For some reason it doesn't work though (still requires V3)
```

Comment:
I think this schema is v5, not v4.
We do not support v5 yet.
Why do you have this schema? Have you seen it in any place?

Reply:
I can't quite remember but I think it's what prod is using (V4). When I tried this schema in the mocks, it didn't work, even after updating the types. You can see in `getProviders.ts` the code shows we are using V4 in prod: 

```js
const getProviders: GetProviders = async () => {
  if (getEnv("MOCK")) return mockGetProviders();

  const res = await network({
    method: "GET",
    url: `${getSwapAPIBaseURL()}/providers`,
    params: { whitelist: getAvailableProviders() },
    paramsSerializer: (params) =>
      qs.stringify(params, { arrayFormat: "comma" }),
  });

  const responseV4 = res.data as ProvidersResponseV4;
  if (!responseV4.providers || !Object.keys(responseV4.providers).length) {
    throw new SwapNoAvailableProviders();
  }
  return Object.entries(responseV4.providers).flatMap(([provider, groups]) => ({
    provider: provider,
    pairs: groups.flatMap((group) =>
      group.methods.flatMap((tradeMethod) =>
        Object.entries(group.pairs).flatMap(([from, toArray]) =>
          (toArray as number[]).map((to) => ({
            from: responseV4.currencies[from],
            to: responseV4.currencies[to.toString()],
            tradeMethod,
          }))
        )
      )
    ),
  }));
};
```

Reply:
Long story short I'm not sure what's going on, but I think we need to understand the different schemas, what's getting used, and then update the mocks. I think another improvement we could make it try to use mocked API responses where appropriate with Playwright HTTP mocking, rather than using `MOCK=1`. To me that's more of an E2E test, and also it reduces the amount of `MOCK` logic in the code.